internal/service/budgets/budget_data_source.go

```diff
@@ -0,0 +1,345 @@
+package budgets
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/budgets"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @SDKDataSource("aws_budgets_budget")
+func DataSourceBudget() *schema.Resource {
+	return &schema.Resource{
+		ReadWithoutTimeout: dataSourceBudgetRead,
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
This could be optional/computed similar to the corresponding budget resource, and just default to the provider account ID when unset.

https://github.com/hashicorp/terraform-provider-aws/blob/baf5cac858ed3379dcb3ffa6aa8a2be2d22fdfab/internal/service/budgets/budget.go#L300-L303